include/highfive/H5Group.hpp

```diff
@@ -10,35 +10,30 @@
 #define HIGHFIVE_H5GROUP_HPP
 
 #include "H5Object.hpp"
+#include "bits/H5_definitions.hpp"
 #include "bits/H5Annotate_traits.hpp"
 #include "bits/H5Node_traits.hpp"
 
 namespace HighFive {
 
-class File;
-
 ///
 /// \brief Represents an hdf5 group
 class Group : public Object,
               public NodeTraits<Group>,
               public AnnotateTraits<Group> {
   public:
-
     const static ObjectType type = ObjectType::Group;
 
-    Group();
-
   protected:
-    explicit Group(const Object& o) : Object(o) {};
+    using Object::Object;
 
-    friend class File;
+    inline Group(Object&& o) noexcept : Object(std::move(o)) {};
 
-    friend class ::HighFive::Reference;
+    friend class File;
+    friend class Reference;
+    template <typename Derivate> friend class ::HighFive::NodeTraits;
 };
-}
 
-#include "bits/H5Node_traits_misc.hpp"
-#include "bits/H5Group_misc.hpp"
```

Comment:
So, I'm interested in this. Thank you for your work! I like the premise of a simplified include structure…

If I were to have a class/namespace/function that only ever takes a `H5Group` and does operations with that group, or extracts `Datasets` from it, will this work? Wouldn't I have to include the implementation somewhere? From my naïve thinking, just including `H5Group.hpp` should allow me to compile a simple library operating only on said group…

Reply:
> If I were to have a class/namespace/function that only ever takes a H5Group and does operations with that group

Internally we know that anything manipulating group must include the implementation.

For the user, in most cases it is a no-problem since the user must have created the group from a File, and then (s)he has all the required header
In the rare case someone wants to do a library or something operating in a group, then he must include `H5File.hpp`. Actually **we probably should update the docs** stating this all new developments should prefer doing the single include of h5File.hpp

Reply:
OK, if this going to be the official policy, I'm fine with that. Thank you for responding!